python/graphstorm/wholegraph/wholegraph.py

```diff
@@ -283,3 +288,241 @@ def load_wg_feat(part_config_path, num_parts, type_name, name):
     feat_wm_embedding.get_embedding_tensor().from_file_prefix(feat_path,
                                                                 part_count=num_parts)
     return feat_wm_embedding
+
+
+def create_wholememory_optimizer(
+    optimizer_type: str, param_dict: dict
+):
+    """Create a wholegraph sparse optimizer.
+
+    If we use wholegraph to store sparse embeddings, for future update, a joint
+    wholegraph sparse optimizer has to be created ahead of time, and then attach
+    to the (wholegraph)sparse embedding.
+
+    Parameters
+    ----------
+    optimizer_type: str
+        optimizer types: [ "sgd" | "adam" | "adagrad" | "rmsprop" ]
+    param_dict: dict
+        parameters of the optimizer
+
+    Returns
+    -------
+    WholeMemoryOptimizer : WholeGraph native optimizer (wgth.WholeMemoryOptimizer)
+    """
+    if wgth is None:
```

Comment:
Shall we check WHOLEGRAPH_INIT instead of wgth?

Reply:
Sounds good.